feature-libs/asm/components/customer-emulation/customer-emulation.component.spec.ts

```diff
@@ -74,11 +114,67 @@ describe('CustomerEmulationComponent', () => {
     fixture.detectChanges();
 
     //Click button
-    const endSessionButton = fixture.debugElement.query(By.css('button'));
+    const endSessionButton = fixture.debugElement.query(
+      By.css('button[formControlName="logoutCustomer"]')
+    );
     spyOn(asmComponentService, 'logoutCustomer').and.stub();
     endSessionButton.nativeElement.click();
 
     //assert
     expect(asmComponentService.logoutCustomer).toHaveBeenCalled();
   });
+
+  it('should assign cart to customer', () => {
```

Comment:
At fist glance, the unit test seems to be written like an e2e test. Can the design be adjusted to test individual behaviours? (like the component doesn't submit the form if the field is empty.).